src/adam_bashforth_adaptive.jl

```diff
@@ -0,0 +1,384 @@
+          # desired. As least one loop is necessary, for the sake of the
+          # adaptive step selection
+          CE_loops = 1
+          i = 0
+          while i<CE_loops && n>=steporder+1
+              i= i+1
+              #(C)orrection of y[n+1]
+              #add new coefficients  based on this prediction
+              ##computation of Φ, Φstar using recurrence relations
+              Φ_coefs!(b, Φ, Φstar,dy[n+1],h,t,n+1,1:steporder+1)
+
+              ##calculation of g_{k}(n) coefficients using recurrence relations
+              ##has already been done above
+
+              ## We correct the computation of y[n+1] using y[n+1]
+              ## as part of our prediction Harier equation 5.7
+              pred = pred + h[n]*g[steporder+1]*Φ[3,steporder+1]
+
+              #(E)valuate again
+              dy[n+1] = F(t[n+1],pred)
+          end
+
+          #If `implicit` == true, we set the corrected prediction to be
+          #y[n+1]. If `implicit` == false, this is simply an explicit solver
+          implicit = true
+          if implicit
+              y[n+1] = pred
+          end
+          #NOTE:fixed step size implicit solver tested and gives correct result
+          ######################################################################
+          #Adaptive step selection
+          ######################################################################
+
+          #Don't use adaptive steps until n >= order + 2
+          if n<steporder+2
+            successful_step = true
+            h[n+1]=min(fac*h[n],(tspan[n+1]-tspan[n])/2) # fac = .8 as default, by convention
+            #h[n+1] = minstep
+          else
+              #Once we have enough initial points, we use adaptive step selection
+
+              #We want to calculate local truncation error, which is differences
+              #between implicit result for order k and k+1. We already have the
+              #result for order k, namely y[n+1]. For order k+1, we would follow same
+              #procedure, and need Φ[n+1,steporder +2] and g[n,steporder + 2]. The
+              #Latter has already been calculated above, the former still needs calculation
+              Φ_coefs!(b, Φ, Φstar,dy[n+1],h,t,n+1,steporder+2)
+
+              #The truncation error , equation right above 7.1 in Harier: le_k+1 = ||y(n+1)_{k+1}-y(n+1)_{k}||
+              diff = (g[steporder+2] - g[steporder+1])*Φ[3,steporder+2]
+              truncation_error = h[n]*diff
+
+              #The book also mentions that the following is approximately the norm of the
+              # truncation error under the assumption of an equidistant grid
+              #h[n]* γ_star(steporder+2)*Φ[n+1,steporder+2]
+
+              # The norm of the truncation error is computed, using the norm
+              # described in eq 4.11 of Harier et al Vol 1, on page 168
+              N = length(y[n+1])
+              sc = Array(Float64,N)
+              errorNorm = 0.0
```

Comment:
Yes, you are right, `error_norm` fits the naming convention better. Though, I'm reworking this part of the code right now, and will likely rename this `err` to fit the naming conventions established in `runge_kutta.jl`. Thanks!